library/x509_crt.c

```diff
@@ -1102,35 +1105,30 @@ int mbedtls_x509_crt_parse_file( mbedtls_x509_crt *chain, const char *path )
     return( ret );
 }
 
-int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
+int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const MBEDTLS_TCHAR *path )
 {
     int ret = 0;
 #if defined(_WIN32) && !defined(EFIX64) && !defined(EFI32)
     int w_ret;
-    WCHAR szDir[MAX_PATH];
-    char filename[MAX_PATH];
-    char *p;
-    size_t len = strlen( path );
+    MBEDTLS_TCHAR szDir[MAX_PATH];
+    MBEDTLS_TCHAR filename[MAX_PATH];
+    MBEDTLS_TCHAR *p;
+    size_t len = mbedtls_strlen( path );
 
-    WIN32_FIND_DATAW file_data;
+    WIN32_FIND_DATA file_data;
     HANDLE hFind;
 
     if( len > MAX_PATH - 3 )
         return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
 
     memset( szDir, 0, sizeof(szDir) );
-    memset( filename, 0, MAX_PATH );
-    memcpy( filename, path, len );
-    filename[len++] = '\\';
+    memset( filename, 0, sizeof(filename) );
+    _tcsncpy( filename, path, len );
```

Comment:
This seems to cause `undefined reference to '_tcsncpy'` when building with MinGW-w64 (v5.0.2)

Reply:
Thanks for testing.
Could you suggest how to define these `_tcslen` and `_tcslen` in MinGW-w64?
In Visual Studio for Unicode builds it is
'#define _tcslen         wcslen'
and
'#define _tcsncpy        wcsncpy'